### PR TITLE
add-tests

### PR DESCRIPTION
**Pull request by TheRealHaui**
Added tests that cover previously uncovered logic.

### CODE REVIEW

File: src/test/java/com/fasterxml/jackson/core/format/DataFormatMatcherTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    MatchStrength matchStrength = MatchStrength.WEAK_MATCH;
+    DataFormatMatcher dataFormatMatcher = new DataFormatMatcher(null,
+            byteArray,
+            0,
+            (-2),
+            null,
+            matchStrength);
+    InputStream inputStream = dataFormatMatcher.getDataStream();
+
+    assertEquals((-2), inputStream.available());
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/core/format/DataFormatMatcherTest.java]: This seems bit odd/wrong actually. `available()` should not, I think, ever return negative value... it's not the next byte (that'd be something like "peek()") but # of bytes that may be read without blocking.

Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/core/format/DataFormatMatcherTest.java]: could you file a bug for this one?

**Comment by cowtowncoder** [comment]
Looks good, and possibly uncovered one minor flaw even!

File: src/test/java/com/fasterxml/jackson/core/format/DataFormatMatcherTest.java (hunk 1 above)
Comment (TheRealHaui) [inline src/test/java/com/fasterxml/jackson/core/format/DataFormatMatcherTest.java]: Sure.

Considered it to be right returning a negative number.
However, read InputStream class comment of the JDK a minute ago and learned that it is not. right.

Comment (TheRealHaui) [inline src/test/java/com/fasterxml/jackson/core/format/DataFormatMatcherTest.java]: Created: https://github.com/FasterXML/jackson-core/issues/391

**Pull request merged**